modules/claims_api/app/sidekiq/claims_api/va_notify_follow_up_job.rb

```diff
@@ -19,10 +24,21 @@ class VANotifyFollowUpJob < ClaimsApi::ServiceBase
       slack_client.notify(msg)
     end
 
-    def perform(notification_id)
+    def perform(notification_id, poa_id) # rubocop:disable Metrics/MethodLength
       status = notification_response_status(notification_id)
       detail = "Status for notification #{notification_id} was '#{status}'"
 
+      # Call logic to map VANotify status to our internal step status
+      step_status = map_notify_status(status)
+      # Update the POA process step with latest status
+      poa = ClaimsApi::PowerOfAttorney.find(poa_id)
+      process = ClaimsApi::Process.find_or_create_by(processable: poa, step_type: 'CLAIMANT_NOTIFICATION')
+      if step_status == 'IN_PROGRESS'
+        process.update!(step_status:, error_messages: [])
+      else
+        process.update!(step_status:, error_messages: [], completed_at: Time.zone.now)
```

Comment:
Let's add a spec around the adding/skipping of completed_at, to ensure that logic or mapping changes in the future won't accidentally change this behavior without us knowing.

Reply:
All set; added specs to cover completed_at logging for retry and non-retry statuses.